rest/server_context_test.go

```diff
@@ -289,3 +291,27 @@ func TestStatsLoggerStopped(t *testing.T) {
 	// sleep a bit to allow the "Stopping stats logging goroutine" debug logging to be printed
 	time.Sleep(time.Millisecond * 10)
 }
+
+func TestObtainManagementEndpointsFromServerContext(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("Test requires Couchbase Server")
+	}
+
+	ctx := NewServerContext(&ServerConfig{})
+	defer ctx.Close()
+
+	eps, err := ctx.ObtainManagementEndpoints()
+	assert.NoError(t, err)
+
+	clusterAddress, _, _, _, _ := ctx.tempConnectionDetails()
+	baseSpec, err := connstr.Parse(clusterAddress)
+	require.NoError(t, err)
+
+	spec, err := connstr.Resolve(baseSpec)
+	require.NoError(t, err)
+
+	for _, httpHost := range spec.HttpHosts {
+		formatted := fmt.Sprintf("http://%s:%d", httpHost.Host, httpHost.Port)
+		assert.Contains(t, eps, formatted)
+	}
```

Comment:
I know this is a test, and works fine in our current setup - but this isn't always true if the bootstrap connstr contains non-existent hosts (which can be the case if a customer scales down their backend).

I'm also not entirely sure there's a direct link between the two, because bootstrap hosts might be hostnames, whereas the cluster could return IP addresses, or vice versa.

Reply:
Think the IP / hostname should be handled by resolve. In terms of existing hosts I'll switch to check at least one match.